snap/info.go

```diff
@@ -43,6 +43,15 @@ type PlugInfo struct {
 	Apps      map[string]*AppInfo
 }
 
+// AppNames returns a list of applications names.
+func (plug *PlugInfo) AppNames() []string {
```

Comment:
Can we please not have that in the public API until we have a more clear reason for it? This same type exposes the "list" of app names via the public map field already, and this is simply iterating over them. We should prefer to use that directly instead.

For the one use case we have, I suggest not even defining this as a function at this time. Simply inline the code.